### ISSUE TITLE
trait(jvm): option to add additional classpath elements

### ISSUE THREAD

**Issue by lburgazzoli** [apache/camel-k#2414]
Assuming one uses the PodTemplate mechanic or define a custom base image with some preloaded maven artifacts or resources, it would be nice to have an option to configure additional classpath elements, like:

```
    kamel run integration.yaml --trait jvm.class-path-entry=/opt/my-artifact.jar
```

**Comment by lburgazzoli** [comment]
@astefanutti @nicolaferraro make sense ?

**Comment by squakez** [comment]
I can work on this.

**Comment by astefanutti** [comment]
That makes sense. Thinking quickly about the custom image use case, I wonder if that could be useful if the image could also provide that kind of information. For example, the operator could configure some traits by querying the container registry API for labels. From the end users standpoint, that would serve the same purpose as modeline I guess...

**Comment by lburgazzoli** [comment]
could be an option, was also wondering if we could have something like the kamel cli config file but for the runtime to be be placed somewhere in the container image, like before starting up the runtime, we process i.e. `/etc/camel/config.yaml` and we act according.

**Comment by astefanutti** [comment]
> could be an option, was also wondering if we could have something like the kamel cli config file but for the runtime to be be placed somewhere in the container image, like before starting up the runtime, we process i.e. `/etc/camel/config.yaml` and we act according.

Right, I thought about it too. The only thing I'm not sure, is that in that case, the operator may have to download all the image layers locally in order to resolve that manifest file. I think it'd be more elegant from an end-user perspective compared to labels, but we have to make sure to clean after ourselves, or maintain a cache.

**Comment by squakez** [comment]
I've just checked it and we already can use a generic configuration (`options`) for this trait to accomplish the goal:
```
kamel run -t jvm.options="-cp /opt/my-artifact.jar" Sample.Java --dev
```
We may introduce an additional `classpath` option to simplify the generic one, though I think that simply documenting the `trait` would be enough.

@astefanutti @lburgazzoli wdyt?

**Comment by astefanutti** [comment]
@squakez I'm not sure if I remember it correctly, does the JVM accept multiple occurrences of `-cp` option? If yes, I agree it could used as a solution, as long as we guarantee _platform_ entries and _user_ entries are properly merged and deduplicated.

**Comment by squakez** [comment]
It does not complain, but the value taken is the last argument, so, it would not work. Also, I did not take in consideration the possible merge or de duplication, that won't be considered if using separate arguments. I'm going to fix this shortly then. Thanks.

**Comment by astefanutti** [comment]
> It does not complain, but the value taken is the last argument, so, it would not work. Also, I did not take in consideration the possible merge or de duplication, that won't be considered if using separate arguments. I'm going to fix this shortly then. Thanks.

Yes, that confirms what I remember from my past experiments.

**Comment by nicolaferraro** [comment]
Would it work if we require that the external image uses a (e.g. "CLASSPATH" or "CAMEL_K_CLASSPATH") env var and we pick it at container startup, inline in `cmd`, merging that with operator deps if needed? (We already have a custom command that prints the actual command)
